job-shop-scheduling/test_job_shop_scheduler.py

```diff
@@ -12,14 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dimod import ExactSolver, BinaryQuadraticModel
+import dwavebinarycsp as dbc
+from dwavebinarycsp.exceptions import ImpossibleBQM
 from itertools import product
 from re import match
+from tabu import TabuSampler
 import unittest
+from unittest.mock import patch
```

Comment:
It's a standard practice to group and order imports like:
- standard library
- 3rd party packages
- local packages

Reply:
Also, for clarity, I much more prefer to see `dwavebinarycsp` than `dbc`, in code (don't have a quote on that :smile:). Rare exceptions to this are a well known packages like `np`, `tf`, `pd`, `nx`. Although, in a case of a few references only I still prefer the expanded/full name.